providers/implementations/keymgmt/mlx_kmgmt.c

```diff
@@ -737,6 +737,21 @@ static void *mlx_kem_dup(const void *vkey, int selection)
         || (ret = OPENSSL_memdup(key, sizeof(*ret))) == NULL)
         return NULL;
 
+    if (ret->propq != NULL
```

Comment:
General what I would do here is set all fields that still need to be copied to NULL right after the memdup. This avoids any issues if returning early and is simple to understand. The MLX_KEY structure should also have a comment on it saying to check the mlx_kem_dup() if any fields are added.

Reply:
This is approved & ready to merge, can we do these changes in a subsequent PR?

Reply:
That's my preference, any refactoring can go in later. With a few different choices available.